int16 trg_x
int16 trg_y
int16 trg_theta